addons/base_import_module/models/ir_module.py

```diff
@@ -258,6 +268,125 @@ def module_uninstall(self):
+                if existing_mod:
+                    mod['state'] = 'installed'
+                else:
+                    mod['state'] = 'uninstalled'
+            if 'module_type' in fields:
+                mod['module_type'] = module_type
+            if 'website' in fields:
+                mod['website'] = f"{APPS_URL}/apps/modules/{major_version}/{module_name}/"
+            if 'app_iframe' in fields:
+                mod['app_iframe'] = '<iframe src="%s/?no_header=1" title="Apps Page"></iframe>' % mod['website']
+        return modules_list
+
+    def button_immediate_install_app(self):
+        module_name = self.env.context.get('module_name')
+        resp = requests.get(
+            f"{APPS_URL}/loempia/download/data_app/{module_name}/{major_version}"
+        )
+        import_module = self.env['base.import.module'].create({
+            'module_file': base64.b64encode(resp.content),
+            'state': 'init',
+            'modules_dependencies': self._get_missing_dependencies(resp.content)
+        })
+
+        return {
+            'name': 'Install an App',
+            'view_mode': 'form',
+            'target': 'new',
+            'res_id': import_module.id,
+            'res_model': 'base.import.module',
+            'type': 'ir.actions.act_window',
+            'context': {'data_module': True}
+        }
+
+    @api.model
+    def _get_missing_dependencies(self, zip_data):
+        fp = BytesIO()
+        fp.write(zip_data)
+        dependencies_to_install = self.env['ir.module.module']
+        known_mods = self.search([])
+        installed_mods = [m.name for m in known_mods if m.state == 'installed']
+        with zipfile.ZipFile(fp, "r") as z:
+            for zf in z.filelist:
+                if zf.file_size > MAX_FILE_SIZE:
+                    raise UserError(_("File '%s' exceed maximum allowed file size", zf.filename))
+            with file_open_temporary_directory(self.env) as module_dir:
+                manifest_files = [
+                    file
+                    for file in z.filelist
+                    if file.filename.count('/') == 1
+                    and file.filename.split('/')[1] in MANIFEST_NAMES
+                ]
+                for manifest in manifest_files:
+                    manifest_path = z.extract(manifest, module_dir)
+                    try:
+                        with file_open(manifest_path, 'rb', env=self.env) as f:
+                            terp = ast.literal_eval(f.read().decode())
+                    except Exception:
+                        continue
+                    unmet_dependencies = set(terp.get('depends', [])).difference(installed_mods)
+                    dependencies_to_install |= known_mods.filtered(lambda m: m.name in unmet_dependencies)
+        description = ''
+        if dependencies_to_install:
+            description = 'The following modules will be also installed:\n'
```

Comment:
translations?

Reply:
When it depends of other apps modules they will be provided in the zip file